violet-framework/src/main/java/com/horstmann/violet/product/diagram/common/node/ColorableNodeWithMethodsInfo.java

```diff
@@ -0,0 +1,138 @@
+package com.horstmann.violet.product.diagram.common.node;
+
+import com.horstmann.violet.product.diagram.abstracts.node.IRenameableNode;
+import com.horstmann.violet.product.diagram.property.text.LineText;
+import com.horstmann.violet.product.diagram.property.text.MultiLineText;
+import com.horstmann.violet.product.diagram.property.text.decorator.OneLineText;
+import com.horstmann.violet.product.diagram.property.text.decorator.RemoveSentenceDecorator;
+import com.horstmann.violet.product.diagram.property.text.decorator.ReplaceSentenceDecorator;
+import com.horstmann.violet.product.diagram.property.text.decorator.UnderlineDecorator;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class ColorableNodeWithMethodsInfo extends ColorableNode implements IRenameableNode {
+
+    public ColorableNodeWithMethodsInfo()
+    {
+        super();
+        methods = new MultiLineText(SIGNATURE_CONVERTER);
+    }
+
+    protected ColorableNodeWithMethodsInfo(ColorableNode node) throws CloneNotSupportedException
+    {
+        super(node);
+        methods = new MultiLineText(SIGNATURE_CONVERTER);
+    }
+
+    @Override
+    public void replaceNodeOccurrences(String oldValue, String newValue) {
+        if (!getMethods().toString().isEmpty()) {
+            MultiLineText renamedMethods = new MultiLineText();
+            renamedMethods.setText(renameMethods(oldValue, newValue));
+            setMethods(renamedMethods);
+        }
+    }
+
+    @Override
+    public LineText getAttributes()
+    {
+        return attributes;
+    }
+
+
+
+    /**
+     *  Finds all of oldValue class occurrences in methods and replaces it with newValue
+     * @param oldValue old class name
+     * @param newValue new class name
+     * @return renamed methods
+     */
+    private String renameMethods(String oldValue, String newValue) {
+        ArrayList<String> methods = new ArrayList<String>(Arrays.asList(getMethods().toEdit().split("\n")));
+        StringBuilder renamedMethods = new StringBuilder();
+        Pattern pattern = Pattern.compile(":\\s*(" + oldValue + ")[,)\\s]*");
+
+        String separator = "";
+        for(String method : methods) {
+            renamedMethods.append(separator + renameOccurrencesInMethod(method, newValue, pattern));
```

Comment:
I suppose it won't work, because I don't want to insert a new line character after the last method. Anyway, I've used iterator instead of for loop, so I get rid of this strange separator variable.